tests/python/LookTest.py

```diff
@@ -1,28 +1,186 @@
-        lk.setProcessSpace("somespace")
-        self.assertEqual("somespace", lk.getProcessSpace())
-        lk.setDescription("this is a test")
-        self.assertEqual("this is a test", lk.getDescription())
-        et = OCIO.ExponentTransform()
-        et.setValue([0.1, 0.2, 0.3, 0.4])
-        lk.setTransform(et)
-        oet = lk.getTransform()
-        vals = oet.getValue()
-        self.assertAlmostEqual(0.2, vals[1], delta=1e-8)
-        iet = OCIO.ExponentTransform()
-        iet.setValue([-0.1, -0.2, -0.3, -0.4])
-        lk.setInverseTransform(iet)
-        ioet = lk.getInverseTransform()
-        vals2 = ioet.getValue()
-        self.assertAlmostEqual(-0.2, vals2[1], delta=1e-8)
+    TEST_PROCESS_SPACES = ['raw', 'lnh', 'vd8', 'a.b.c.', '1-2-3-']
+
+    def setUp(self):
+        self.look = OCIO.Look()
+
+    def tearDown(self):
+        self.look = None
+
+    def test_name(self):
+        """
+        Test the setName() and getName() methods.
+        """
+
+        # Default initialized name value is ""
+        self.assertEqual(self.look.getName(), '')
+
+        for name in TEST_NAMES:
+            self.look.setName(name)
+            self.assertEqual(name, self.look.getName())
+
+        # Wrong type tests.
+        for invalid in (None, 1):
+            with self.assertRaises(TypeError):
+                self.look.setName(invalid)
+
+    def test_process(self):
+        """
+        Test the setProcessSpace() and getProcessName() methods.
+        """
+
+        # Default initialized process space value is ""
+        self.assertEqual(self.look.getProcessSpace(), '')
+
+        for process_space in self.TEST_PROCESS_SPACES:
+            self.look.setProcessSpace(process_space)
+            self.assertEqual(process_space, self.look.getProcessSpace())
+
+        # Wrong type tests.
+        for invalid in (None, 1):
+            with self.assertRaises(TypeError):
+                self.look.setProcessSpace(invalid)
+
+    def test_description(self):
+        """
+        Test the setDescription() and getDescription() methods.
+        """
+
+        # Default initialized description value is ""
+        self.assertEqual(self.look.getDescription(), '')
+
+        for desc in TEST_DESCS:
+            self.look.setDescription(desc)
+            self.assertEqual(desc, self.look.getDescription())
+
+        # Wrong type tests.
+        for invalid in (None, 1):
+            with self.assertRaises(TypeError):
+                self.look.setDescription(invalid)
+
+    def test_transform(self):
+        """
+        Test the setTransform() and getTransform() methods.
+        """
+
+        # Default initialized transform value is None
+        self.assertIsNone(self.look.getTransform())
+
+        exp_tr = OCIO.ExponentTransform()
+        exp_values = [0.1, 0.2, 0.3, 0.4]
+        exp_tr.setValue(exp_values)
+        self.look.setTransform(exp_tr)
+        oet = self.look.getTransform()
```

Comment:
[minor] Maybe use `out_exp_tr` instead of `oet` for clarity? Could do this in the inverse test method too.

Reply:
Resolved in 2b67585.